### PR TITLE
Add support for tosa argmax operation

### PR DESCRIPTION
**Pull request by simon-camp**
Fixes #291.

### CODE REVIEW

File: test/Conversion/tosa-to-emitc.mlir [hunk 1]
```diff
@@ -238,6 +238,12 @@ func @test_matmul(%arg0: tensor<1x14x19xf32>, %arg1: tensor<1x19x28xf32>) -> ten
 }
 
 // Reduce ops
+func @test_argmax(%arg0: tensor<13x21x3xi32>) -> tensor<21x3xi32> {
```
Comment (marbre) [inline test/Conversion/tosa-to-emitc.mlir:241]: NIT: This isn't a reduce op, thus the test should be inserted before the comment above (or the comment should be moved).

**Review by marbre — CHANGES_REQUESTED** [review]

File: lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp [hunk 2]
```diff
@@ -606,8 +606,9 @@ class ReduceOpConversion : public OpConversionPattern<SrcOp> {
   using OpConversionPattern<SrcOp>::OpConversionPattern;
 
 public:
-  ReduceOpConversion(MLIRContext *ctx, StringRef funcName)
-      : OpConversionPattern<SrcOp>(ctx), funcName(funcName) {}
+  ReduceOpConversion(MLIRContext *ctx, StringRef funcName, bool keepDims)
```
Comment (dajuro) [inline lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp:609]: Is a reduction without dropping reduced dimensions still a reduction?

File: test/Conversion/tosa-to-emitc.mlir (hunk 1 above)
Comment (marbre) [reply test/Conversion/tosa-to-emitc.mlir:240]: To be more precise, `// Reduce ops` here only refers to ops that have `reuduce` in their name.

**Comment by marbre** [comment]
@dajuro and @heidebro can one of you explicitly look at the function? I only had a look to the mechanics.

@simon-camp please also add a note to https://github.com/iml130/mlir-emitc/blob/main/docs/tosa-op-coverage.md :)

**Review by dajuro — APPROVED** [review]

**Comment by marbre** [comment]
> Fixes #291.

Should have been #279. #291 links to this merge request :wink:

**Pull request merged**